### PR TITLE
012 add new role

### PR DESCRIPTION
**Pull request by Patrick-Mc-Grath**

### CODE REVIEW

File: src/main/java/com/kainos/ea/service/JobRoleService.java [hunk 1]
```diff
@@ -13,13 +17,36 @@ public class JobRoleService {
...
+
+    public int createRole(JobRoleRequest jobRoleRequest) throws FailedToCreateJobRoleException, InvalidJobRoleException {
+        try {
+            String validation = jobRoleValidator.isValidJobRole(jobRoleRequest);
+            if(validation != null){
+                throw new InvalidJobRoleException(validation);
+            }
+            int id = jobRoleDao.createRole(jobRoleRequest, databaseConnector.getConnection());
+            int rolebandid = jobRoleDao.createRoleBand(id, jobRoleRequest.getBandId(), databaseConnector.getConnection());
+            System.out.println(rolebandid);
```
Comment (ccurran01) [inline src/main/java/com/kainos/ea/service/JobRoleService.java:40]: Could this be removed?

File: src/main/java/com/kainos/ea/validator/JobRoleValidator.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.kainos.ea.validator;
+
+import com.kainos.ea.model.JobRoleRequest;
+
+public class JobRoleValidator {
+    public String isValidJobRole(JobRoleRequest jobRoleRequest) {
```
Comment (ccurran01) [inline src/main/java/com/kainos/ea/validator/JobRoleValidator.java:6]: Can we check that the role and band id are defined here too

File: src/test/java/com/kainos/ea/integration/JobRolesIntegrationTest.java [hunk 3]
```diff
@@ -39,4 +45,35 @@ void getEmployees_shouldReturnListOfEmployees()  {
 
         Assertions.assertEquals(200, response.get().getStatus());
     }
+  
+   @Test
+    void postJobRole_shouldReturnIdOfNewlyCreatedJobRole() {
```
Comment (ccurran01) [inline src/test/java/com/kainos/ea/integration/JobRolesIntegrationTest.java:50]: Nice, could we write some SQL here to delete this from the DB?

**Review by ccurran01 — COMMENTED** [review]
Looks good @Patrick-Mc-Grath

File: src/main/java/com/kainos/ea/validator/JobRoleValidator.java (hunk 2 above)
Comment (ccurran01) [reply src/main/java/com/kainos/ea/validator/JobRoleValidator.java:6]: And a minimum length check

**Review by ccurran01 — APPROVED** [review]

File: src/test/java/com/kainos/ea/integration/JobRolesIntegrationTest.java [hunk 4]
```diff
@@ -33,10 +50,47 @@ void getEmployees_shouldReturnListOfEmployees()  {
         
         Assertions.assertTrue(jobRoles.size() > 0);
 
-        Assertions.assertEquals(1, pojos.get(1).getRoleId());
+        Assertions.assertEquals(29, pojos.get(1).getRoleId());
```
Comment (Patrick-Mc-Grath) [inline src/test/java/com/kainos/ea/integration/JobRolesIntegrationTest.java:53]: Made a mistake while trying to see if i could facilitate the delete methods in the dao, so the first role that gets returned here is 29

**Pull request merged**